idaes_examples/notebooks/docs/unit_models/custom_unit_models/creating_unit_model.ipynb

```diff
@@ -0,0 +1,2813 @@
+    "Let's commence by importing the essential libraries from pyomo and idaes."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "8f1d9869",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Import Pyomo libraries\n",
+    "from pyomo.common.config import ConfigBlock, ConfigValue, In, Bool\n",
+    "from pyomo.environ import (\n",
+    "    value,\n",
+    "    Constraint,\n",
+    "    check_optimal_termination,\n",
+    ")\n",
+    "\n",
+    "# Import IDAES cores\n",
+    "from idaes.core import (\n",
+    "    ControlVolume0DBlock,\n",
+    "    declare_process_block_class,\n",
+    "    MaterialBalanceType,\n",
+    "    EnergyBalanceType,\n",
+    "    MaterialFlowBasis,\n",
+    "    MomentumBalanceType,\n",
+    "    UnitModelBlockData,\n",
+    "    useDefault,\n",
+    ")\n",
+    "from idaes.core.util.config import (\n",
+    "    is_physical_parameter_block,\n",
+    "    is_reaction_parameter_block,\n",
+    ")\n",
+    "\n",
+    "import idaes.logger as idaeslog\n",
+    "from idaes.core.solvers import get_solver\n",
+    "from idaes.core.util.model_statistics import degrees_of_freedom\n",
+    "from idaes.core.util.exceptions import ConfigurationError, InitializationError"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8ba94ad6",
+   "metadata": {},
+   "source": [
+    "## 3.2 Creating the unit model\n",
+    "\n",
+    "Creating a unit model starts by creating a class called `LiqExtractionData` and use the `declare_process_block_class` decorator. The `LiqExtractionData` inherts the properties of `UnitModelBlockData` class, which allows us to create a control volume which is necessary for the unit model. After declaration of the class we proceed to define the relevant config arguments for the control volume. The config arguments includes the following properties:\n",
+    "\n",
+    "- `material_balance_type` - Indicates what type of mass balance should be constructed\n",
+    "- `has_pressure_change` - Indicates whether terms for pressure change should be\n",
+    "constructed\n",
+    "- `has_phase_equilibrium` - Indicates whether terms for phase equilibrium should be\n",
+    "constructed\n",
+    "- `Organic Property` - Property parameter object used to define property calculations\n",
+    "for the Organic phase\n",
+    "- `Organic Property Arguments` - Arguments to use for constructing Organic phase properties\n",
+    "- `Aqueous Property` - Property parameter object used to define property calculations\n",
+    "for the aqueous phase\n",
+    "- `Aqueous Property Arguments` - Arguments to use for constructing aqueous phase properties\n",
+    "\n",
+    "As there are no pressure changes or reactions in this scenario, configuration arguments for these aspects are not included. However, additional details on configuration arguments can be found [here](https://github.com/IDAES/idaes-pse/blob/8948c6ce27d4c7f2c06b377a173f413599091998/idaes/models/unit_models/cstr.py)."
```

Comment:
The URL links to the CSTR model, is this the correct reference?

Reply:
I believe this is a good example which has most of the config and also includes reaction property package and enthalpy flow terms and hence thought it would be a good place. I am not sure if this is the best example or not.